oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticRequestHandler.java

```diff
@@ -169,14 +169,13 @@ public BoolQueryBuilder baseQuery() {
                 for (IndexDefinition.IndexingRule r : elasticIndexDefinition.getDefinedRules()) {
                     sp.addAll(r.getSimilarityProperties());
                 }
+                String mltQueryString = propertyRestrictionQuery.replace("mlt?", "");
```

Comment:
Hm, this replaces all occurrences... So if the query is "mtl?abcmtl?" then the result is "abc" instead of "abcmtl?". So I think we should use

    propertyRestrictionQuery.replaceFirst("^mtl\\?", "");